docs/Fabulous.XamarinForms/views-perf.md

```diff
@@ -47,6 +47,39 @@ You can also use
 * the `fix` function for portions of a view that have no dependencies at all (besides the "dispatch" function)
 * the `fixf` function for command callbacks that have no dependencies at all (besides the "dispatch" function)
 
+### View keys 101 
+  1. What is view key? 
+      View key is a little helper property like in this sample: 
+      
+     ```fsharp
+         // Previous View
+         View.Grid([
+                 View.Label(key = "header", text = "Old Header")
+                 View.Label(key = "body", text = "Old Body")
+             ])
+         
+         // New View
+         View.Grid([
+             View.Label(key = "header", text = "New Header") // Will reuse previous header
+             View.Button(key = "body", text = "New body") // Won't be able to reuse previous body since Label != Button
+         ])
+     ```
+         
+  As we can see here, we have added a 'key' parameter to each Label.
+  Key should be unique across view elements in same collection, i.e.
+  Children of StackLayout and so on.
+  
+  2.What is a purpose of key? 
+   Purpose of key is following: 
+  * make determination of proper view element to replace easier(just by key,yes)
+  * most important,to keep a reference to previous XF control(so we would not create it, but reuse existing).
```

Comment:
I think we should explain the purpose of `key` from the user's point of view here.

When might the view not be performant?
=> Anything changing the ordering of controls from one update to the other: reordering, adding/removing elements at the start, etc.

Why is it a bad thing?
=> Fabulous will try to reuse existing controls as much as possible, in order by default. Changing the ordering might force Fabulous to recreate the controls which might impact the app.

How can I prevent that?
=> You can give intent to controls with the `key` property. Fabulous will try to use those intents to match controls having the same intent between updates, reusing controls more effectively.

For example:
```fs
View.StackLayout([
    if model.ShowFirstVideo then
        yield View.MediaElement(source = MediaPath "path/to/video.mp4")

    yield View.MediaElement(key = "other-video", source = MediaPath "path/to/other-video.mp4")
    yield View.Button(text = "Toggle first video", command = (fun () -> dispatch ToggleFirstVideo))
])
```

In this case, when `ShowFirstVideo` = false, the StackLayout will have 2 children, the 1st and 2nd video players.
When `ShowFirstVideo` = true, there will be only 1 child left, the 2nd video player.

Due to how Fabulous reuses views between updates, when switching from `ShowFirstVideo` true to false, Fabulous will remove the 2nd video player and reuse the 1st video player (the source will be the 2nd path), which might lose the current state of the 2nd player if it was playing.
That's because Fabulous has no way of knowing the real intent behind your code.

You can help it by specifying the `key` property which will let Fabulous know that the 2nd video player should still be there after the update.
Fabulous will only remove the 1st video player and keep the 2nd video player.